src/ast/rewriter/seq_rewriter.cpp

```diff
@@ -3218,6 +3470,12 @@ br_status seq_rewriter::mk_re_star(expr* a, expr_ref& result) {
         result = re().mk_star(re().mk_union(b1, c1));
         return BR_REWRITE2;
     }
+    expr *a1 = nullptr, *a2 = nullptr, *cond = nullptr;
```

Comment:
The intent with adding the ite-lifting in mk_app_core is to factor out the special cases from each operator.

Reply:
Yes indeed, I only left these special cases in unary operations, as that avoids the term blowup which you pointed out. These were no longer necessary for the ITE lifting, I just left them in as useful rewrites more generally.